ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/TableRowDiffy.scala

```diff
@@ -47,48 +48,54 @@ class TableRowDiffy(tableSchema: TableSchema,
       .parseAndClose(new StringReader(schemaString), classOf[TableSchema])
 
   // scalastyle:off cyclomatic.complexity
-  private def diff(x: Record, y: Record,
+  private def diff(x: Option[Record], y: Option[Record],
                    fields: Seq[TableFieldSchema], root: String): Seq[Delta] = {
-    def getField(f: String)(x: Record): AnyRef = {
-      x.get(f)
+    def getField(f: String)(x: Record): Option[AnyRef] = {
+      Option(x.get(f))
     }
 
     fields.flatMap { f =>
       val name = f.getName
       val fullName = if (root.isEmpty) name else root + "." + name
       if (f.getType == "RECORD" && f.getMode != "REPEATED") {
-        val a = x.get(name).asInstanceOf[Record]
-        val b = y.get(name).asInstanceOf[Record]
-        if (a == null && b == null) {
+        val a = x.flatMap(r => getField(name)(r).map(_.asInstanceOf[Record]))
+        val b = y.flatMap(r => getField(name)(r).map(_.asInstanceOf[Record]))
+        if (a.isEmpty && b.isEmpty) {
           Nil
-        } else if (a == null || b == null) {
-          Seq(Delta(fullName, Option(a), Option(b), UnknownDelta))
+        } else if (a.isEmpty || b.isEmpty) {
+          Seq(Delta(fullName, a, b, UnknownDelta))
         } else {
           diff(a, b, f.getFields.asScala, fullName)
         }
       } else if (f.getMode == "REPEATED" && unordered.contains(fullName)) {
-        if (f.getType == "RECORD" && unorderedFieldKeys.contains(fullName) &&
-            unordered.exists(_.startsWith(s"$fullName."))) {
-          val a = sortList(x.get(name).asInstanceOf[java.util.List[Record]],
-            unorderedFieldKeys.get(fullName).map(getField))
-          val b = sortList(y.get(name).asInstanceOf[java.util.List[Record]],
-            unorderedFieldKeys.get(fullName).map(getField))
-          a.asScala.zip(b.asScala).flatMap{case (l, r) =>
-            diff(l, r, f.getFields.asScala, fullName)
-          }
-        }
-        else {
-          val a = sortList(x.get(name).asInstanceOf[java.util.List[AnyRef]])
-          val b = sortList(y.get(name).asInstanceOf[java.util.List[AnyRef]])
-          if (a == b) Nil else Seq(Delta(fullName, Option(a), Option(b), delta(a, b)))
+        if (f.getType == "RECORD"
+          && unorderedFieldKeys.contains(fullName)
+          && unordered.contains(fullName)) {
```

Comment:
this is also checked above already; similar comments in this file to other files